src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -667,6 +668,44 @@ public override ImmutableArray<string> TupleElementNames
         }
 
         /// <summary>
+        /// Get the fields for the tuple's elements (in order and cached).
+        /// </summary>
+        public override ImmutableArray<FieldSymbol> TupleElementFields
+        {
+            get
+            {
+                if (_lazyFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyFields, CollectTupleElementFields());
+                }
+
+                return _lazyFields;
+            }
+        }
+
+        private ImmutableArray<FieldSymbol> CollectTupleElementFields()
+        {
+            var builder = ArrayBuilder<FieldSymbol>.GetInstance(_elementTypes.Length);
+            builder.SetItem(_elementTypes.Length - 1, null);
+            var members = GetMembers().Where(m => m.Kind == SymbolKind.Field);
```

Comment:
Is this going to allocate an intermediate array? Whenever is possible, we avoid Linq and lambdas in compilers. This filtering can easily be done in the loop below, it should just loop through GetMembers().